source/codegen/metadata_validation.py

```diff
@@ -6,6 +6,18 @@
 from schema import And, Optional, Or, Schema, Use  # type: ignore
 
 
+_ALREADY_MUTATED = (
```

Comment:
This should have both a technical and a historical coment. Something like "grpc device metadata mutations have been applied to MI drivers in source to minimize transformations that happen outside of hapigen"

Would it also be fair to say that it "makes sense" that all drivers would do this? Not that we should make the change, but  that this is probably a better model than what we're doing for the other drivers. At least for metadata that is generated/expanded/transformed by another process (like `hapigen` or `scrapigen`)

Reply:
That comments sounds good. Added it.

And yes, I think it makes sense to do the mutations wherever the metadata is sourced (hapigen, scrapigen, or here for hand-written ones). It feels better and more transparent for the process creating the metadata to produce the final version used in grpc-device. Yeah, I'd say it would make sense to do the mutations for scrapigen-sourced drivers as well eventually.